### ISSUE TITLE
Is there a flag for being able to read XMP from an apparently protected PDF file?

### ISSUE THREAD

**Issue by amtopel** [adobe/xmp-docs#13]
Sometimes my XMP code cannot read from certain PDF files.  Then, when I open the PDF file and do some sort of edit to it (like add a comment or something), my XMP code will be able to read.  I'm wondering if the edit changes the security of the PDF.  At any rate, is there any way for my code to read the metadata from the apparently locked or secured PDF file?  Thanks for any help.

Just to clarify, OpenFile is working, but GetXMP is not (it's returning false).

**Comment by angupt** [comment]
@amtopel As per GetXMP documentation, **api returns true if any XMP is present in file otherwise false**. I dont think just editing will change the security of your pdf file. As per my understanding, when you open/edit your pdf file, your pdf opener(e.g Acrobat) might be writing some XMP  metadata in the pdf file like last modified date etc. that's why after editing you are able to fetch XMP metadata.

**Comment by amtopel** [comment]
Sorry, I wasn't really clear and didn't know my exact problems. Please let me rephrase my question.  I've come across several PDFs recently that I open successfully, but return FALSE when GetXMP is called.  They'll only return TRUE once I've opened the PDF in Acrobat Reader, for example, and made some sort of edit.  Do you know why that might be?  In general, why does GetXMP ever return FALSE?
________________________________

**Comment by angupt** [comment]
**I've come across several PDFs recently that I open successfully, but return FALSE when GetXMP is called.**  This might be because the original file does not contain any xmp metadata and GETXMP returns false when file does have any XMP metadata.

**They'll only return TRUE once I've opened the PDF in Acrobat Reader, for example, and made some sort of edit.**   This is because when you open/edit pdf file in acrobat or similar tools, they might insert some metadata in it like modification history and if you call GetXMP api on these files, it returns true because now they contain XMP metadata.

**In general, why does GetXMP ever return FALSE?** One possible reason when file doesn't have any XMP metadata.

**P.S. you can verify metadata of pdf file in acrobat by right click->Document Properties->Description Tab**

**Comment by amtopel** [comment]
Thank you very much for your responses.  So can I do programmatically what Acrobat is likely doing?  In other words, if GetXMP returns FALSE, is there something my code can do to add metadata so that it works?  It seems circular, because I thought the only way to add metadata with XMP toolkit would be after calling GetXMP, but if GetXMP returns false because there’s no metadata, then what can you do?

**Comment by angupt** [comment]
@amtopel GetXMP api returns the file metadata only. It cannot be used to add or update metadata in file. For adding/updating metadata in file, XMP toolkit provide PutXMP api . Please go through the documentation of PutXMP api under **"XMP-Toolkit-SDK-CC201607\XMP-Toolkit-SDK-CC201607\public\include\TXMPFiles.hpp"**

**Comment by papitaawale** [comment]
Hi,
I want to check xmp support for reading metadata for password protected pdf file

does xmp supports reading metadata for password protected pdf fiel, if yes then how?
if no, then is there any other API which supports the reading of metadata for password protected pdf file? (i want to do it programmatically using c++ code + i want free sdk/api)
